src/plugins/intel_cpu/src/graph.cpp

```diff
@@ -830,6 +830,31 @@ void Graph::AllocateWithReuse() {
     }
 
     if (!undefinedBoxes.empty()) {
+        // Use proxy memory manager for output edges
+        for (auto& box : undefinedBoxes) {
+            for (auto& edge : edge_clusters[box.id]) {
+                const auto child = edge->getChild();
+                if (edge->getStatus() == Edge::Status::NeedAllocation &&
+                    child->getType() == Type::Output) {
+                    auto proxyMemMngr =
+                        std::make_shared<ProxyMemoryMngr>(std::make_shared<DnnlMemoryMngr>(make_unique<MemoryMngrWithReuse>()));
+                    DEBUG_LOG(proxyMemMngr, " ", this);
+
+                    // Store the output memory managers.
+                    // So that, the infer requests can be able to access them.
+                    int count = 0;
+                    for (auto &output : outputNodesMap) {
+                        if (output.second == child) {
+                            outputNodesMemMngrMap[output.first] = proxyMemMngr;
+                            count++;
+                        }
+                    }
+                    IE_ASSERT(count == 1);
+                }
+            }
+        }
+        IE_ASSERT(outputNodesMemMngrMap.size() <= outputNodesMap.size());
```

Comment:
Please add meaningful error message for easier debugging in future.

Reply:
unnecessary. removed.